sdk/common/src/main/java/com/mesosphere/sdk/offer/CommonIdUtils.java

```diff
@@ -3,17 +3,14 @@
 import java.util.UUID;
```

Comment:
Is it possible for us to wind up in the scenario of:
1. User upgrades to new version
2. Task gets relaunched with new style task id
3. User downgrades to old version
4. We then somehow try to regenerate the task name for the relaunched task and are trying to generate one with out the "other stuff" prefix.

Reply:
That is a good question.

Reply:
@nickbp: Can we write some test(s) that prove this out one way or another?

Reply:
We only retrieve the task name in a couple places. One is to figure out the goal state from a TaskStatus and the other is to validate during initial launch that the task id isn't malformed (i.e. we don't actually use the name value).

In practice the task id is generated from the task spec from scratch at launch time. It just gets regenerated each time since the UUID is supposed to change.

That said I could still do a ServiceTest or something if it's desirable

Reply:
OK, yeah this really shouldn't be an issue now that I think about it more. Task names are generated from the service specs. Where TaskInfos are concerned, task name is a separate field from task id. The task id -> task name extraction only really comes up with TaskStatuses, where there is no task name field. Hence the above case where we're figuring out the goal state from a task status (e.g. to figure out if 'FINISHED' is the desired state, after a task has finished).